### PR TITLE
Post refactor - Selector

### PR DESCRIPTION
**Pull request by cbellot000**

### CODE REVIEW

File: src/ansys/dpf/post/tools.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+tools
+=====
+This module holds factories to create geometry, selections...
+These factories help creating Objects used to defined which results are evaluated.
+"""
+
+from ansys.dpf.post import selection
+
+
+def select(time_freq_indexes=None, time_freq_sets=None, time_freq_values=None, named_selection_names=None, kwargs=None):
```
Comment (cbellot000) [inline src/ansys/dpf/post/tools.py:11]: named_selection_name

File: src/ansys/dpf/post/tools.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    """Creates a ``Selection`` instance allowing to choose the domain on which results are evaluated.
+    The results domain defines the time frequency and the spatial selection.
+
+    Parameters
+    ----------
+
+    """
+    current_selection = selection.Selection()
```
Comment (cbellot000) [inline src/ansys/dpf/post/tools.py:19]: parse inputs and raise

File: src/ansys/dpf/post/selection.py [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+            raise TypeError(f"The input scoping is an instance of {str(type(scoping))} "
+                            f"instead of an expected {str(Scoping)}.")
+
+        op = operators.utility.forward(
+            scoping,
+            server=self._server
+        )
+        self._selection.add_operator(op)
+        self._selection.set_output_name(_WfNames.scoping, op.outputs.any)
+
```
Comment (cbellot000) [inline src/ansys/dpf/post/selection.py:242]: select_body(partid)

File: src/ansys/dpf/post/selection.py [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+        elements : list[int], Scoping
+            element Ids.
+        """
+        if isinstance(elements, Scoping):
+            scoping = elements
+        else:
+            scoping = Scoping(location=locations.elemental, ids=elements, server=self._server)
+        self.select_with_scoping(scoping)
+
+    def select_intersection_with(self, spatial_selection) -> None:
```
Comment (cbellot000) [inline src/ansys/dpf/post/selection.py:273]: intersect(...)

File: src/ansys/dpf/post/selection.py [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+    ----------
+
+
+    """
+
+    def __init__(self, server=None):
+        self._server = server
+        self._spatial_selection = SpatialSelection(server=server)
+        self._time_freq_selection = TimeFreqSelection(server=server)
+        self._qualifier_selection = None
```
Comment (germa89) [inline src/ansys/dpf/post/selection.py:35]: I think we should be instantiating the object the properties itself. Like

```suggestion
    def __init__(self, time_freq_select=None, nodes=Node, server=None):
        self._server = server
        self._spatial_selection = SpatialSelection(nodes=nodes, server=server)
        self._time_freq_selection = TimeFreqSelection(time_freq_select=time_freq_select, server=server)
        self._qualifier_selection = None
```

**Review by germa89 — CHANGES_REQUESTED** [review]

**Comment by PProfizi** [comment]
Closing as changes have been integrated in #208

**Pull request closed without merging**